### PR TITLE
Error out if we find an unknown command-line argument.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This also got auto-formatted out from under me! Sorry about the whitespace changes.

### CODE REVIEW

File: support/entry/entry.cpp [hunk 1]
```diff
@@ -154,6 +152,24 @@ struct CommandLineArgs {
...
+  std::cerr << "  -separate-present             Prefers a separate present queue" << std::endl;
+  std::cerr << "  -output-frame=<frame>         Dumps the given frame to a file an exits" << std::endl;
+  std::cerr << "  -load-pipeline-cache=<file>   Loads and uses a pipeline cache from the given location" << std::endl;
+  std::cerr << "  -write-pipeline-cache=<file>  Writes the applicaitons pipeline cache to the given location" << std::endl;
+  std::cerr << "  -shader-compiler=<string>     Sets the shader compiler to the given one, if the sample could use multiple" << std::endl;
+  std::cerr << "  -validation                   Turns on the validation layers if available" << std::endl;
+  std::cerr << "  -output-file                  Sets the output file for the output-frame argument" << std::endl;
+  std::cerr << "  -wait-for-debugger            Forces the application to pause on starup until a debugger is attached" << std::endl;
+  std::cerr << "  -help                         Print this help" << std::endl;
+  exit(-1);
```
Comment (ndeakin) [inline support/entry/entry.cpp]: I doubt this is terribly important, but I think it's atypical linux program behavior for `prog -help` to exit with failure. Perhaps we can have `prog -help` return success, for example, by leaving the caller of print_usage to decide what to return.

File: support/entry/entry.cpp [hunk 2]
```diff
@@ -167,42 +183,39 @@ void parse_args(CommandLineArgs* args, int argc, const char** argv) {
...
+    } else if (strncmp(argv[i], "-help", 5) == 0) {
+      print_usage(argv);
+    } else {
+      std::cerr << "Unknown command line argument " << argv[i] << std::endl;
+      print_usage(argv);
     }
   }
 }
+
+
```
Comment (ndeakin) [inline support/entry/entry.cpp]: nit: extra newline?

File: support/entry/entry.cpp [hunk 3]
```diff
@@ -361,13 +375,15 @@ int main(int argc, const char** argv) {
...
+      while (!entry::k_stream_started) {
+      }  // Wait for the stream to be started
       return_value = main_entry(&entry_data);
     });
     std::atomic<bool> exited(false);
     std::thread ggp_thread([&exited]() {
-      while(!exited) {
-        while(ggp_event_queue->ProcessEvent()) {}
+      while (!exited) {
+        while (ggp_event_queue->ProcessEvent()) {
```
Comment (ndeakin) [inline support/entry/entry.cpp]: really minor nit: this style change appears different from how empty functions are handled (based on a change above) - not sure if intentional or not so just figured I'd mention it.

Comment (AWoloszyn) [reply support/entry/entry.cpp]: I defer to clang-format for all my formatting needs!

**Pull request merged**